runelite-client/src/main/java/net/runelite/client/chat/ChatMessageManager.java

```diff
@@ -122,7 +122,15 @@ public void onSetMessage(SetMessage setMessage)
 			case TRADE:
 			case AUTOCHAT:
 			case PUBLIC:
-				usernameColor = isChatboxTransparent ? chatColorConfig.transparentUsername() : chatColorConfig.opaqueUsername();
+				boolean isFriend = client.isFriended(setMessage.getName(), true) && !client.getLocalPlayer().getName().equals(setMessage.getName());
```

Comment:
Does this normally say you are your own friend? I would guess that is impossible.

Reply:
I thought so too, I'll open a ticket so I can explain it in a bit

Reply:
#3694